addOns/retire/src/test/java/org/zaproxy/addon/retire/model/VersionPatternUnitTest.java

```diff
@@ -38,15 +42,15 @@ private static Stream<Arguments> getFilenameTestValuesAndExpectedVersions() {
                 Arguments.of("bootstrap-3_1_1.min.js", "3_1_1"),
                 Arguments.of("bootstrap-3-1-1.min.js", "3-1-1"),
                 Arguments.of("bootstrap-3.1.1.js", "3.1.1"),
-                Arguments.of("bootstrap-3.1.1-alpha.js", "3.1.1-alpha"),
```

Comment:
This is not ideal (e.g. `1.4.0-beta.2` is an in use/valid up to version).

Reply:
I'd have to look again but I'm pretty sure that the version checking code isn't handling alpha/beta/build, etc

Reply:
We should change/fix that otherwise we'll be reporting FPs again.

Reply:
But the handling is likely project/library dependant. Ex how does alpha/beta compare vs "next" or "rc" or "build" or "b"....

Reply:
We should look at what retire.js is doing.

Reply:
Per: https://github.com/RetireJS/retire.js/blob/68d2b47c12f6cd2cd569d645852692ca87727ad2/repository/README.md?plain=1#L11 it is meant to capture alpha/beta/rc1, etc. However, per the following the "strings" aren't handled/compared:
- https://github.com/search?q=repo%3ARetireJS%2Fretire.js+beta&type=code
- https://github.com/search?q=repo%3ARetireJS%2Fretire.js+alpha&type=code

Reply:
https://www.baeldung.com/java-comparing-versions Maybe ComparableVersion is a good way to go… 🤔

Reply:
IMO we should do the same as:
https://github.com/RetireJS/retire.js/blob/68d2b47c12f6cd2cd569d645852692ca87727ad2/node/lib/retire.js#L121-L141

Reply:
Pretty sure that is what we're doing. Will double check.

Reply:
Looks like it.

Reply:
Though that code handles beta/alpha well.

Reply:
I don't see it, but I'll test and revise.

Reply:
It does lexicographical comparison, beta > alpha.

Reply:
Okay the existing code does handle lexigraphic comparison.

<details>
<summary>Here's my current suggestion</summary>

```diff
diff --git a/addOns/retire/src/main/java/org/zaproxy/addon/retire/RetireUtil.java b/addOns/retire/src/main/java/org/zaproxy/addon/retire/RetireUtil.java
index c461ec6..1ed236f 100644
--- a/addOns/retire/src/main/java/org/zaproxy/addon/retire/RetireUtil.java
+++ b/addOns/retire/src/main/java/org/zaproxy/addon/retire/RetireUtil.java
@@ -21,6 +21,7 @@
 
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
 import java.util.Formatter;
 import org.apache.commons.httpclient.URI;
 import org.apache.commons.httpclient.URIException;
@@ -79,7 +80,23 @@
      *  version1(of a particular JS library) is >= version2(of a particular JS library)
      */
     public static boolean isAtOrAbove(String version1, String version2) {
-        String[] v1 = version1.split("[\\.-]");
+        String[] v1 = version1.split("[\\._-]");
+        if (v1.length == 1) {
+            // There were no separators in the "version" being checked
+            // Likely a cache buster string, ex: 7a06f256
+            return true;
+        }
+        int isAllZeros = 9;
+        try {
+            isAllZeros = Arrays.stream(v1).mapToInt(Integer::parseInt).sum();
+        } catch (NumberFormatException nfe) {
+            // Nothing to do, it might happen
+        }
+        if (isAllZeros == 0) {
+            // Probably matched empty guid, ex: 00000000-0000-0000-0000-000000000000
+            // Yes zero is always "below" some version
+            return true;
+        }
         String[] v2 = version2.split("[\\.-]");
         int l = v1.length > v2.length ? v1.length : v2.length;
         for (int i = 0; i < l; i++) {
```

</details>